### PR TITLE
Polkadot extra for type

### PR DESCRIPTION
**Pull request by henri-ly**

### CODE REVIEW

File: src/api/polkadot/bisontrails.js [hunk 1]
```diff
@@ -13,6 +13,43 @@ const agent = new https.Agent({
 
 const LIMIT = 100;
 
+const getExtra = (type, extrinsic) => {
```
[the remark this answers is not in the document]
Comment (rjarasson-ledger) [reply src/api/polkadot/bisontrails.js:16]: Don't know if there is possibly some extras available though...

File: src/api/polkadot/common.js [hunk 2]
```diff
@@ -14,17 +14,17 @@ export const getOperationType = (pallet: string, palletMethod: string) => {
...
+      return "NOMINATE";
 
     case "chill":
-      return "UNDELEGATE";
-  
+      return "CHILL";
+
     case "withdraw_unbonded":
     case "withdrawUnbonded":
       return "WITHDRAW_UNBONDED";
```
[the remark this answers is not in the document]
Comment (rjarasson-ledger) [reply src/api/polkadot/common.js:30]: no, payout stakers gives absolutely no rewards, it's a GENERIC-kind of operation.

File: src/api/polkadot/bisontrails.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (henri-ly) [reply src/api/polkadot/bisontrails.js:16]: Well I don't have the data available but will probably update if i found the data

File: src/api/polkadot/subscan.js [hunk 3]
```diff
@@ -15,6 +15,50 @@ const SUBSCAN_MULTIPLIER = 10000000000;
...
+      break;
+
+    case "UNBOND":
+      extra = {
+        ...extra,
+        unbondedAmount: BigNumber(params.value),
+      };
+      break;
+
+    case "REWARD":
```
Comment (rjarasson-ledger) [inline src/api/polkadot/subscan.js:41]: If REWARD is handled with amount, SLASH should also have amount in  extras.

File: src/api/polkadot/subscan.js [hunk 4]
```diff
@@ -15,6 +15,50 @@ const SUBSCAN_MULTIPLIER = 10000000000;
...
+      extra = {
+        ...extra,
+        validatorStash: params.find((p) => p.type === "AccountId").value,
```
Comment (rjarasson-ledger) [inline src/api/polkadot/subscan.js:44]: value are not decoded via subscan, so the encoding part is missing here.

File: src/api/polkadot/bisontrails.js [hunk 5]
```diff
@@ -13,6 +13,43 @@ const agent = new https.Agent({
...
+      break;
+
+    case "UNBOND":
+      extra = {
+        ...extra,
+        unbondedAmount: BigNumber(extrinsic.amount),
+      };
+      break;
+
+    case "REWARD":
```
Comment (rjarasson-ledger) [inline src/api/polkadot/bisontrails.js:33]: Should add SLASH then

File: src/api/polkadot/bisontrails.js [hunk 6]
```diff
@@ -13,6 +13,43 @@ const agent = new https.Agent({
...
+      extra = {
+        ...extra,
+        validatorStash: extrinsic.validatorStash,
```
Comment (rjarasson-ledger) [inline src/api/polkadot/bisontrails.js:36]: no Amount like for Subscan ?

File: src/api/polkadot/subscan.js [hunk 7]
```diff
@@ -102,10 +143,12 @@ const mapSubscanExtrinsic = (
...
     senders: recipient ? [addr] : [],
     recipients: recipient ? [recipient] : [],
-    extra: {
-      module: extrinsic.call_module,
-      function: extrinsic.call_module_function,
-    },
+    extra: getExtra(
+      type,
+      extrinsic,
+      extrinsic.call_module,
```
Comment (rjarasson-ledger) [inline src/api/polkadot/subscan.js:149]: Can you put this logic into getExtra directly, since extrinsic is already passed to function ?

**Review by rjarasson-ledger — CHANGES_REQUESTED** [review]

File: src/api/polkadot/subscan.js [hunk 8]
```diff
@@ -15,6 +15,50 @@ const SUBSCAN_MULTIPLIER = 10000000000;
...
+};
+
+const getExtra = (type, extrinsic, pallet, method) => {
+  let extra = {
+    palletMethod: `${pallet}.${method}`,
+  };
+
+  const params = JSON.parse(extrinsic.params);
+
+  switch (type) {
```
[the remark this answers is not in the document]
Comment (henri-ly) [reply src/api/polkadot/subscan.js:29]: not sure, information doesn't mean anything to me : 
`params: "[{"name":"num_slashing_spans","type":"u32","value":0,"value_raw":""}]"`

File: src/api/polkadot/subscan.js (hunk 3 above)
Comment (henri-ly) [reply src/api/polkadot/subscan.js:41]: good point

File: src/api/polkadot/subscan.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (haammar-ledger) [reply src/api/polkadot/subscan.js:29]: Nah it's not that.
:rip: for subscan, maybe from BT later :man_shrugging:

File: src/api/polkadot/bisontrails.js (hunk 5 above)
Comment (henri-ly) [reply src/api/polkadot/bisontrails.js:33]: I'll redo it when we got the data structure from BT

File: src/api/polkadot/bisontrails.js (hunk 6 above)
Comment (henri-ly) [reply src/api/polkadot/bisontrails.js:36]: I will wait for the next release of the API then redo a PR with the correct data

**Comment by henri-ly** [comment]
Close PR because i did bad with the rebase

**Pull request closed without merging**